http-clients/aws-crt-client/src/it/java/software/amazon/awssdk/http/crt/AwsCrtCombinatorialConfigStressIntegrationTest.java

```diff
@@ -0,0 +1,213 @@
+            }
+        }
+    }
+
+    @DataPoints("EventLoop")
+    public static int[] eventLoopValues(){
+        return new int[]{1, 4};
+    }
+
+    @DataPoints("ConnectionPool")
+    public static int[] connectionsValues(){
+        /* Don't use 1 connection Pool of size 1, otherwise test takes too long */
+        return new int[]{10, 100};
+    }
+
+    @DataPoints("NumRequests")
+    public static int[] requestValues(){
+        return new int[]{1, 25, 250};
+    }
+
+    @DataPoints("ParallelClients")
+    public static int[] parallelClientValues(){
+        return new int[]{1, 2, 8};
+    }
+
+    @DataPoints("SharedClient")
+    public static boolean[] sharedClientValue(){
+        return new boolean[]{true, false};
+    }
+
+    @Theory
+    public void checkAllCombinations(@FromDataPoints("EventLoop") int eventLoopSize,
+                                     @FromDataPoints("ConnectionPool") int connectionPoolSize,
+                                     @FromDataPoints("NumRequests") int numberOfRequests,
+                                     @FromDataPoints("ParallelClients") int numberOfParallelClients,
+                                     @FromDataPoints("SharedClient") boolean useSharedClient) throws Exception {
+
+        try {
+            if (CrtResource.getAllocatedNativeResourceCount() > 0) {
+                System.err.println("Leaked Resources: " + String.join(", ", CrtResource.getAllocatedNativeResources()));
+            }
+            Assert.assertEquals("Expected Zero allocated AwsCrtResources", 0, CrtResource.getAllocatedNativeResourceCount());
+
+            String testName = String.format("Testing with eventLoopSize %d, connectionPoolSize %d, numberOfRequests %d, " +
+                            "numberOfParallelJavaClients %d, useSharedClient %b", eventLoopSize, connectionPoolSize,
+                    numberOfRequests, numberOfParallelClients, useSharedClient);
+            System.out.println("\n" + testName);
+
+            CountDownLatch latch = new CountDownLatch(numberOfParallelClients);
+
+            AttributeMap attributes = AttributeMap.builder()
+                    .put(SdkHttpConfigurationOption.MAX_CONNECTIONS, connectionPoolSize)
+                    .build();
+
+
+            SdkAsyncHttpClient awsCrtHttpClient = AwsCrtAsyncHttpClient.builder()
+                    .eventLoopSize(eventLoopSize)
+                    .buildWithDefaults(attributes);
+
+            KmsAsyncClient sharedAsyncKMSClient = KmsAsyncClient.builder()
+                    .region(REGION)
+                    .httpClient(awsCrtHttpClient)
+                    .credentialsProvider(CREDENTIALS_PROVIDER_CHAIN)
+                    .build();
+
+            final AtomicBoolean failed = new AtomicBoolean(false);
+
+            long start = System.currentTimeMillis();
+            ExecutorService pool = Executors.newCachedThreadPool();
+            for (int threads = 0; threads < numberOfParallelClients; threads++) {
```

Comment:
Why are we doing this in multiple threads? The clients are already running in the even loop group.

Reply:
We want to cover all possible user calling patterns and make sure that we don't see any crashes with those calling patterns. Customers may have multiple threads sharing a single `AwsCrtAsyncHttpClient` so we added that case to this test.